errors.go

```diff
@@ -335,3 +335,29 @@ func Find(origErr error, test func(error) bool) error {
 	})
 	return foundErr
 }
+
+// Chain ties together two errors, giving them a Causer interface and a concatenated Error message.
+type chain struct {
```

Comment:
need override  `ErrorStack` or `%+v` to print errors info before wrap?

e.g.

```go
func main() {
	_, err := time.ParseDuration("dfasdf")
	err = resetErrDataTooLong("c", 1,  err)
	err = resetErrDataTooLong("b", 2, err)
	fmt.Println(errors.ErrorStack(err))
}

func resetErrDataTooLong(colName string, rowIdx int, err error) error {
	newErr := types.ErrDataTooLong.Gen("Data too long for column '%v' at row %v", colName, rowIdx)
	return errors.Wrap(err, newErr)
}
```

should output:

```
time: invalid duration dfasdf
/home/robi/Code/go/src/github.com/pingcap/tidb/executor/test/zz.go:24: [types:1406]Data too long for column 'c' at row 1
/home/robi/Code/go/src/github.com/pingcap/tidb/executor/test/zz.go:24: [types:1406]Data too long for column 'b' at row 2
```
I think `time: invalid duration dfasdf` is what we need for this method

Reply:
Okay, the formatter is added.